compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BObjectType.java

```diff
@@ -72,6 +92,85 @@ public <T, R> R accept(BTypeVisitor<T, R> visitor, T t) {
         return visitor.visit(this, t);
     }
 
+    private boolean hasTypeHoles() {
+        return fields.values().stream().anyMatch(field -> field.type instanceof BNoType);
+    }
+
+    @Override
+    public SemType semType() {
+        if (od != null) {
+            return od.getSemType(env);
+        }
+        od = new ObjectDefinition();
+        // I don't think this is actually possible
+        assert !hasTypeHoles() : "unimplemented";
+        List<Member> members = new ArrayList<>(fields.size());
+        ObjectQualifiers qualifiers = getObjectQualifiers();
+        Set<String> memberNames = new HashSet<>();
+        for (BField field : fields.values()) {
+            Optional<Member> member = createMember(field, qualifiers.readonly(), memberNames);
+            member.ifPresent(members::add);
+        }
+
+        BObjectTypeSymbol objectSymbol = (BObjectTypeSymbol) this.tsymbol;
+        for (BAttachedFunction fun : objectSymbol.attachedFuncs) {
+            Optional<Member> member = createMember(fun, memberNames);
+            member.ifPresent(members::add);
+        }
+        return od.define(env, qualifiers, members);
```

Comment:
@heshanpadmasiri Do we need to consider `distinct` objects? Can't see the logic here.

Reply:
Fixed with #[1f3e8e6](https://github.com/ballerina-platform/ballerina-lang/pull/43010/commits/1f3e8e6e14af4273e5ba06156a92f7b05ea31532), However due to #43129 we don't get the correct flag in `BObject` type at the moment